server/etcdmain/config.go

```diff
@@ -236,6 +236,7 @@ func newConfig() *config {
 	fs.StringVar(&cfg.ec.Logger, "logger", "zap", "Currently only supports 'zap' for structured logging.")
 	fs.Var(flags.NewUniqueStringsValue(embed.DefaultLogOutput), "log-outputs", "Specify 'stdout' or 'stderr' to skip journald logging even when running under systemd, or list of comma separated output targets.")
 	fs.StringVar(&cfg.ec.LogLevel, "log-level", logutil.DefaultLogLevel, "Configures log level. Only supports debug, info, warn, error, panic, or fatal. Default 'info'.")
+	fs.StringVar(&cfg.ec.LogRotateConfigJSON, "log-rotate-config-json", "", "Configures log rotation with a JSON lumberjack logger config. Supports a single file output target.")
```

Comment:
Can we at least define/re-define the JSON format in the etcd documentation? Our users should not need to learn `lumberjack`. Also, shall we provide a default config? Probably we have to divide the flag into two parts, one to enable the rotation and the other one to config it.

Reply:
this makes sense.